Fixed support for custom activity dialog fragment

jkoryf opened the pull request:
Fixes #35

jkoryf committed:
Fixed support for custom activity dialog fragment

emaccaull commented on app/src/main/java/com/paranoidandroid/journey/legplanner/fragments/CustomActivityCreatorFragment.java:
```diff
@@ -73,8 +75,19 @@ public static CustomActivityCreatorFragment newInstance(LatLng coordinates, Date
+    // For compatibility we should include the deprecated method too
+    @Override
+    public void onAttach(Activity context) {
+        System.out.println("context is "+context);
```
I think this is debug code? Can you use Log.d() if you want to keep it?

jkoryf replied:
The PR is now updated to remove this line

emaccaull commented on app/src/main/java/com/paranoidandroid/journey/legplanner/fragments/CustomActivityCreatorFragment.java:
```diff
@@ -73,8 +75,19 @@ public static CustomActivityCreatorFragment newInstance(LatLng coordinates, Date
+
     @Override
     public void onAttach(Context context) {
+        System.out.println("context is "+context);
```
Same here.

jkoryf replied:
The PR is now updated to remove this line

emaccaull commented on app/src/main/java/com/paranoidandroid/journey/legplanner/activities/PlannerActivity.java:
```diff
@@ -383,12 +383,10 @@ public void addFromBookmarksPressed(View v) {
         Day day = getDayPlannerFragment().getSelectedDay();
         Destination dest = day.getLeg().getDestination();
-        FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentManager fragmentManager = getFragmentManager();
```
Is it safe to use this when inheriting from AppCompatActivity? I remember Nathan saying it could cause problems to use getFragmentManager() in this case.

jkoryf replied:
I believe it's OK to use android.app.FragmentManager if the fragments that you are going to load with it are native fragments (i.e. android.app.Fragment). 
I had to implement the CustomActivityCreatorFragment dialog as an android.app.DialogFragment instead of its support equivalent, in order to support the PlaceAutocompleteFragment inside. 
Its equivalent support option, SupportPlaceAutocompleteFragment, was having a lot of issues during my tests on API 19.

jkoryf committed:
Correcting for PR comments
Removed debug logging
Corrected google map api key meta-data in AndroidManifest.xml

emaccaull reviewed (approved):
:+1:  Looks good!
